kfdefs/overlays/osc/osc-cl2/trino/configs/catalogs/kafka_fx.properties

```diff
@@ -0,0 +1,3 @@
+connector.name=kafka
+kafka.table-names=FX
+kafka.nodes=my-cluster-kafka-brokers.kafka.svc.cluster.local:9092
```

Comment:
@bryonbaker & @caldeirav I'm guessing we need more here to allow trino to authenticate against the kafka cluster, please advise on what properties need to be added.

Reply:
This stream is a bit of a proof of concept. It is wide open to anyone on the cluster so no auth needed. We will add that as a second pass.
I a not sure you have set the kafka node correctly - shouldn't it be the bootstrap server? (see test below)

You can test connectivity to the kafka topic with:
oc run kafka-consumer -ti --image=[quay.io/strimzi/kafka:latest-kafka-3.0.0](http://quay.io/strimzi/kafka:latest-kafka-3.0.0) --rm=true --restart=Never -- bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap.kafka.svc.cluster.local:9092 --topic FX

Reply:
Updated, @bryonbaker can you take another look.

Reply:
Looks good! Thanks

Reply:
I think there is another step to go for the table definition file. I laeft a comment on this issue: https://github.com/os-climate/os_c_data_commons/issues/193#issuecomment-1193601420

Repeated here for convenience:

<repeat>
@HumairAK - thanks very much. I think there is one thing left. There is a json file that needs to be mounted in trino. I had the link above to the file above. I have converted that to a config map (see below). That needs to be mounted at the correct place for Trino to find.

The documentation indicates that we need a property to describe where the table definition files are: https://trino.io/docs/current/connector/kafka.html#table-definition-files

Can we get permissions set up so I can edit this config map?

The config map we want loaded is:

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: trino-schema
data:
  trino-schema.json: |-
    {
        "tableName": "FX",
        "schemaName": "tpch",
        "topicName": "tpch.FX",
        "key": {
            "dataFormat": "json",
            "fields": [
                {
                    "name": "currency",
                    "type": "VARCHAR",
                    "hidden": "false"
                },
                {
                    "name": "base_currency",
                    "type": "VARCHAR",
                    "hidden": "false"
                }
            ]
        },
        "message": {
            "dataFormat" :"json",
            "fields" : [
                {
                    "name": "currency",
                    "mapping": "currency",
                    "type": "VARCHAR"
                },
                {
                    "name": "base_currency",
                    "mapping": "base_currency",
                    "type": "VARCHAR"
                },
                {
                    "name": "ask",
                    "mapping": "ask",
                    "type": "DOUBLE"
                },
                {
                    "name": "bid",
                    "mapping": "bid",
                    "type": "DOUBLE"
                },
                {
                    "name": "date",
                    "mapping": "date",
                    "type": "TIMESTAMP",
                    "dataFormat" :"custom-date-time",
                    "formatHint" : "E M d HH:mm:ss z yyyy"
                },
                {
                    "name": "   high_ask",
                    "mapping": "high_ask",
                    "type": "DOUBLE"
                },
                {
                    "name": "high_bid",
                    "mapping": "high_bid",
                    "type": "DOUBLE"
                },
                {
                    "name": "low_ask",
                    "mapping": "low_ask",
                    "type": "DOUBLE"
                },
                {
                    "name": "low_bid",
                    "mapping": "low_bid",
                    "type": "DOUBLE"
                },
                {
                    "name": "midpoint",
                    "mapping": "midpoint",
                    "type": "DOUBLE"
                }
            ]
        }
    }
```
</repeat>